File: split.js
const fs = require('fs');

const html = fs.readFileSync('index.html', 'utf8');

const styleRegex = /<style>([\s\S]*?)<\/style>/i;
const styleMatch = html.match(styleRegex);

const scriptRegex = /<script>([\s\S]*?)<\/script>/g;
let scripts = [];
let match;
while ((match = scriptRegex.exec(html)) !== null) {
  scripts.push(match[1]);
}

if (styleMatch) {
  fs.writeFileSync('css/style.css', styleMatch[1].trim());
}

if (scripts.length > 0) {
  fs.writeFileSync('js/app.js', scripts.join('\n\n').trim());
}

let newHtml = html.replace(styleRegex, '<link rel="stylesheet" href="css/style.css">');
newHtml = newHtml.replace(/<script>[\s\S]*?<\/script>/g, '');
newHtml = newHtml.replace('</body>', '    <script src="https://cdnjs.cloudflare.com/ajax/libs/three.js/r128/three.min.js"></script>\n    <script src="https://cdnjs.cloudflare.com/ajax/libs/vanilla-tilt/1.8.1/vanilla-tilt.min.js"></script>\n    <script src="js/app.js"></script>\n</body>');

fs.writeFileSync('index.html', newHtml);
console.log('Done splitting!');

File: split.py
import re

with open('index.html', 'r', encoding='utf-8') as f:
    html = f.read()

style_match = re.search(r'<style>([\s\S]*?)</style>', html, re.IGNORECASE)
if style_match:
    with open('css/style.css', 'w', encoding='utf-8') as f:
        f.write(style_match.group(1).strip())

script_matches = re.findall(r'<script>([\s\S]*?)</script>', html)
if script_matches:
    with open('js/app.js', 'w', encoding='utf-8') as f:
        f.write('\n\n'.join(script_matches).strip())

new_html = re.sub(r'<style>[\s\S]*?</style>', '<link rel="stylesheet" href="css/style.css">', html, count=1, flags=re.IGNORECASE)
new_html = re.sub(r'<script>[\s\S]*?</script>', '', new_html)
new_html = new_html.replace('</body>', '    <script src="https://cdnjs.cloudflare.com/ajax/libs/three.js/r128/three.min.js"></script>\n    <script src="https://cdnjs.cloudflare.com/ajax/libs/vanilla-tilt/1.8.1/vanilla-tilt.min.js"></script>\n    <script src="js/app.js"></script>\n</body>')

with open('index.html', 'w', encoding='utf-8') as f:
    f.write(new_html)

print('Done splitting!')
